v22.1/api-support-policy.md

```diff
@@ -0,0 +1,108 @@
+---
+title: API Support Policy
+summary: Learn about Cockroach Labs' policy for supporting CockroachDB APIs.
+toc: true
+docs_area: reference
+---
+
+Cockroach Labs exposes various application programming interfaces (APIs).
+
+The vast majority of changes to these interfaces are seamless additions of new functionality. However, some changes are backward-incompatible and may require you to adjust your integration. Changes to an API are introduced according to its *stability guarantees*, which depend on its *support policy*.
+
+This page includes the following information:
+
+- Our [support policies](#support-policies) for each type of API.
+- Our [stability guarantees](#stability-guarantees) for supported APIs.
+- A summary of [APIs](#apis) that CockroachDB makes available.
+
+## Support policies
+
+| Type             | Description                                                               | Stability guarantees |
+|------------------|---------------------------------------------------------------------------|----------------------|
+| Programmable     | Supported for interfacing with third-party automated tools.               | ✓                    |
+| Non-programmable | Supported for consumption by humans. Not supported for automation.        |                      |
+| Reserved         | Intended for use by CockroachDB developers. Not supported for public use. |                      |
+
+### Stability guarantees
+
+An API is *stable* if it does not receive backward-incompatible changes. API stability implies programmability.
+
+The following stability guarantees apply to **programmable** APIs. These APIs remain stable within major versions:
+
+- Backward-incompatible changes may be introduced in new major versions.
+- Backward-compatible changes may be introduced in new minor versions.
+
+Stability guarantees do **not** apply to non-programmable and reserved APIs:
+
+- Backward-incompatible changes may be introduced in new minor and major versions.
+
+#### API changes
+
+The following types of API changes qualify as *backward-incompatible* (or "breaking changes"):
+
+- Removal of an endpoint, SQL statement, [built-in function](functions-and-operators.html#built-in-functions), [cluster setting](cluster-settings.html), or session variable.
+- Addition or removal of a mandatory or optional command-line flag.
```

Comment:
This bullet feels potentially overloaded. Do we really want to say that the addition of an optional command-line flag is backward-incompatible? I could see the removal of such a flag as backward-incompatible, but not the addition. Thoughts? In case we agree here, I'd suggest we split this into two bullets:

- Addition or removal of a mandatory command-line flag.
- Removal of an optional command-line flag.